locksmith/src/operations/pricingOperations.ts

```diff
@@ -0,0 +1,173 @@
+import networks from '@unlock-protocol/networks'
+import { getFees, getGasCost } from '../utils/pricing'
+import * as lockSettingOperations from './lockSettingOperations'
+import { MIN_PAYMENT_STRIPE } from '../utils/constants'
+
+interface Price {
+  decimals: number
+  symbol: string
+  price: number
+  timestamp: number
+  confidence: number
+  creditCardEnabled: boolean
+}
+
+export interface LockPricingValues {
+  total: number
+  subtotal: number
+  gasCost: number
+  unlockServiceFee: number
+  creditCardProcessingFee: number
+  isCreditCardPurchasable: boolean
+}
+
+export interface LockPricingProps {
+  amount?: number
+  address?: string
+  lockAddress?: string
+  keysToPurchase?: number
+  network: number
+}
+
+export type PriceResults = Partial<
+  Price & {
+    priceInAmount: number
+  }
+>
+
+export async function getDefiLammaPrice({
+  network,
+  address,
+  amount = 1,
+}: Omit<LockPricingProps, 'lockAddress'>): Promise<PriceResults> {
+  const networkConfig = networks[network]
+
+  if (!network) {
+    return {}
+  }
+
+  const items: string[] = []
+  const coingecko = `coingecko:${networkConfig.nativeCurrency?.coingecko}`
+  const mainnetTokenAddress = networkConfig.tokens?.find(
+    (item) => item.address?.toLowerCase() === address?.toLowerCase()
+  )?.mainnetAddress
+
+  if (mainnetTokenAddress) {
+    items.push(`ethereum:${mainnetTokenAddress}`)
+  }
+
+  if (address) {
+    items.push(`${networkConfig.chain}:${address}`)
+  }
+
+  if (!address && coingecko) {
+    items.push(coingecko)
+  }
+
+  const endpoint = `https://coins.llama.fi/prices/current/${items.join(',')}`
+  const response = await fetch(endpoint)
+
+  if (!response.ok) {
+    return {}
+  }
+
+  const json: Record<'coins', Record<string, Price>> = await response.json()
+  const item = Object.values(json.coins).filter(
+    (item) => item.confidence > 0.95
+  )[0]
+
+  if (!item) {
+    return {}
+  }
+
+  const priceInAmount = item.price * amount
+
+  return {
+    ...item,
+    priceInAmount,
+  }
+}
+
+/**
+ * Get lock pricing with priority to credit card total if present
+ * @returns
+ */
+export async function getUsdPricingForLock({
+  lockAddress,
+  network,
+  address,
+  amount = 1,
+  keysToPurchase = 1,
+}: LockPricingProps): Promise<PriceResults> {
+  // priority to credit card if present settings is present
+  if (lockAddress) {
+    const { creditCardPrice } = await lockSettingOperations.getSettings({
+      lockAddress,
+      network,
+    })
+
+    if (creditCardPrice) {
+      return {
+        confidence: 1,
+        priceInAmount: creditCardPrice * keysToPurchase,
```

Comment:
You need to pass a few more things to be consistent with defilamma pricing output here.

Reply:
actually non needed the return type is 

```
export type PriceResults = Partial<
  Price & {
    priceInAmount: number
  }
>
```

and is the same for `getUsdPricingForLock` and `getDefiLammaPrice`

Reply:
That is only if it cannot get the price which is why it is partial otherwise if it can find the price, it will return all so you do need to provide.

Reply:
updated!